Fix the CONF_LOOP check to use the config

automaton82 opened the pull request:
The CONF_LOOP check was not using the config value but rather just the string.

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Inside the Environment Canada camera, it checks CONF_LOOP as a string but it's supposed to check the config value associated with this key to see if it loops or not.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml
camera:
  - platform: environment_canada
    latitude: 43.6532
    longitude: -79.3832
    name: Weather Camera
    loop: false
```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #38889
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-asc+-review%3Aapproved

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

automaton82 committed:
Fix the CONF_LOOP check to use the config

The CONF_LOOP check was not using the config value but rather just the string.

MatthewFlamm commented:
Your PR is still failing the black check, despite you checking that you did it.  Maybe you forgot to commit and push those changes after running it?

Changes look good to me.

automaton82 commented:
Yes is passes on my local black linter so I don't know what's going on. I admit I'm not overly familiar with git just yet, I suspect something is messing up. Mostly use TFS professionally.

I'll take another crack at it tomorrow and try another pull.

MatthewFlamm commented:
This is what it wants if you want to do it manually:

```
-    add_devices([ECCamera(radar_object, config.get(CONF_NAME), config.get(CONF_LOOP))], True)
+    add_devices(
+        [ECCamera(radar_object, config.get(CONF_NAME), config.get(CONF_LOOP))], True
+    )
```

automaton82 committed:
Attempting to manually fix the Black check

I'm having problems with VSCode commit at the moment so I'm trying to fix the file raw to pass the Black lint test like it does on my local machine.

automaton82 commented:
Yea I tried that now by editing the file raw on Github to be exactly that but it's giving the same error. I'll have to try again locally

automaton82 commented:
Oh wait nevermind, it did work. It passed now.

michaeldavie commented:
Thanks for catching this. I tested it on the `dev` branch and it looks good to me.

automaton82 commented:
Thanks! Still trying to get tox to run so I can pass the local tests...

MatthewFlamm commented:
What tests are you trying to run?  I don't see any environment_canada tests.  I don't think you have to run the local tests in this case.

automaton82 commented:
Oh okay. I was just trying to run all of them because that's what that checkbox required in the pull request.. it says it won't be merged without being checked.

MatthewFlamm commented:
It says it won't merge unless the tests **pass**.  You can also invoke pytest directly https://developers.home-assistant.io/docs/development_testing#testing-outside-of-tox

automaton82 commented:
Thanks! I did end up getting it working and they do appear to pass. I'll check the box.

MartinHjelmare commented on homeassistant/components/environment_canada/camera.py:
```diff
@@ -52,18 +52,21 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
-    add_devices([ECCamera(radar_object, config.get(CONF_NAME))], True)
+    add_devices(
+        [ECCamera(radar_object, config.get(CONF_NAME), config.get(CONF_LOOP))], True
```
Please use `dict[key]` for required schema keys and keys with default schema values.
```suggestion
        [ECCamera(radar_object, config.get(CONF_NAME), config[CONF_LOOP])], True
```

automaton82 replied:
Sure, it's been changed.

automaton82 committed:
Changing to use dict[key] as per recommendation.

MartinHjelmare reviewed (approved):
Thanks!
